backend-rust/src/graphql_api.rs

```diff
@@ -1160,26 +1160,57 @@ pub struct SubscriptionContext {
 impl SubscriptionContext {
     const ACCOUNTS_UPDATED_CHANNEL: &'static str = "account_updated";
     const BLOCK_ADDED_CHANNEL: &'static str = "block_added";
+    const RETRY_DELAY_SEC: u64 = 5;
 
     pub async fn listen(self, pool: PgPool, stop_signal: CancellationToken) -> anyhow::Result<()> {
-        let mut listener = sqlx::postgres::PgListener::connect_with(&pool)
+        loop {
+            match self.run_listener(&pool, &stop_signal).await {
+                Ok(_) => {
+                    info!("PgListener stopped gracefully.");
+                    break; // Graceful exit, stop the loop
+                }
+                Err(err) => {
+                    println!("err: {:?}", err);
+                    error!("PgListener encountered an error: {}. Retrying...", err);
+
+                    // Check if the stop signal has been triggered before retrying
+                    if stop_signal.is_cancelled() {
+                        info!("Stop signal received. Exiting PgListener loop.");
+                        break;
+                    }
+
+                    tokio::time::sleep(std::time::Duration::from_secs(Self::RETRY_DELAY_SEC)).await;
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    async fn run_listener(
+        &self,
+        pool: &PgPool,
+        stop_signal: &CancellationToken,
+    ) -> anyhow::Result<()> {
+        let mut listener = sqlx::postgres::PgListener::connect_with(pool)
             .await
-            .context("Failed to create a postgreSQL listener")?;
+            .context("Failed to create a PostgreSQL listener")?;
 
         listener
             .listen_all([Self::BLOCK_ADDED_CHANNEL, Self::ACCOUNTS_UPDATED_CHANNEL])
             .await
-            .context("Failed to listen to postgreSQL notifications")?;
+            .context("Failed to listen to PostgreSQL notifications")?;
 
         let exit = stop_signal
             .run_until_cancelled(async move {
                 loop {
+                    println!("Listening for notifications...");
```

Comment:
Should be removed if it was for debugging this PR or if kept then `tracing::info()` might be better.